core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -26,56 +26,95 @@ import scala.util.control.NonFatal
+  case object UpdateGlobalJobExecGauges
 
   def props(executionServiceCluster: ExecutionServiceCluster,
             datasource: SlickDataSource,
+            bucketCredential: Credential,
             submissionPollInterval: FiniteDuration = 1 minutes,
             workbenchMetricBaseName: String): Props = {
-    Props(new SubmissionSupervisor(executionServiceCluster, datasource, submissionPollInterval, workbenchMetricBaseName))
+    Props(new SubmissionSupervisor(executionServiceCluster, datasource, bucketCredential, submissionPollInterval, workbenchMetricBaseName))
   }
 }
 
 /**
  * Supervisor actor that should run for the life of the app. SubmissionStarted messages will start a monitor
  * for the given submission. Errors are logged if that monitor fails.
- * 
+ *
  * @param executionServiceCluster
  * @param datasource
  * @param submissionPollInterval
  */
 class SubmissionSupervisor(executionServiceCluster: ExecutionServiceCluster,
                            datasource: SlickDataSource,
+                           bucketCredential: Credential,
                            submissionPollInterval: FiniteDuration,
                            override val workbenchMetricBaseName: String) extends Actor with LazyLogging with RawlsInstrumented {
   import context._
 
-  // These fields are marked volatile because it is read by a separate statsd thread.
-  // It is only written by this actor.
+  /* A note on metrics:
+   *  We instrument four things here:
+   *  1. The number of submissions in each state across the whole system.
+   *  2. The number of workflows in each state across the whole system.
+   *  3. The number of submissions in each state, by workspace, for any workspace with at least one running submission.
+   *  4. The number of workflows in each state, by workspace and submission, for any submission with at least one running workflow.
+   *
+   *  Aggregating over #3 and #4 will give you strange results, as metrics will disappear when submissions or workspaces finish running.
+   *  #3 and #4 are meant to be used as a way to diagnose the behaviour of a particular workspace or submission.
+   *
+   *  Every SubmissionMonitorActor calculates #3 and sends it in the SaveCurrentWorkflowStatusCounts message. Therefore two submissions
+   *  running in the same workspace will update activeSubmissionStatusCounts to the same value once per poll interval. This is silly but fine.
+   *
+   *  The fields below are marked volatile because they are read by a separate statsd thread.
+   *  They are only written by this actor.
+   */
   @volatile
-  private var currentWorkflowStatusCounts: Map[WorkflowStatus, Int] = Map.empty
+  private var activeSubmissionStatusCounts: Map[WorkspaceName, Map[SubmissionStatus, Int]] = Map.empty
   @volatile
-  private var currentSubmissionStatusCounts: Map[SubmissionStatus, Int] = Map.empty
+  private var activeWorkflowStatusCounts: Map[UUID, Map[WorkflowStatus, Int]] = Map.empty
```

Comment:
Proposal: maybe we should disable these metrics at the submission level. We're over our limit in Hosted Graphite and it looks like submission-level metrics account for a lot of it:

https://www.hostedgraphite.com/accounts/metrics/delete/preview/?pattern=*

We can keep the code that collects the data; maybe just disable creating the gauges in `initGauge`. WDYT?

Also it looks like there are some issues with the URI-redaction logic; I'll look at addressing those in a separate PR.

Reply:
I can add a config flag for creating these metrics, but I'd like to look into aging them out first.

Reply:
ok, I can look into more the aging-out feature of hosted graphite.